### ISSUE TITLE
Theme accessibility: Images seem to be incorrectly assigned the role of "button" and interfering with screen reader access to content

### ISSUE THREAD

**Issue by pauljacobson** [Automattic/wp-calypso#94809]
### Quick summary

This is a report from a customer who uses a screen reader. As I understand the customer's report, images are assigned the role of "button" where these images should be decorative. The effect of this is that when the customer browses a page with their screen reader, it is "snared" by these images while trying to read the actual content.

This produces what seems to be a fairly disruptive experience for the customer and others using screen readers.

It doesn't seem like there is an option for the user to change this in the editor.

### Steps to reproduce

1. Using an unspecified screen reader, move through a site using the letter "G" for "graphic"
2. Notice that a number of images are assigned the role "button" and are set to "Keyboard-focusable"
3. This causes the screen reader to highlight them as clickable where they are not, and should not be

### What you expected to happen

More "decorative" images should be assigned the role of "image" and not be set to "Keyboard-focusable", like this:

![CleanShot 2024-09-23 at 10 29 32@2x](https://github.com/user-attachments/assets/012eadbe-8ea2-470e-b0fc-f8c6a331d5b2)

This is an Image Block added to a template in the Twenty Twenty-Four theme.

### What actually happened

Instead, a number of images that should be "decorative" and be assigned the role of "image", are assigned the role of "button" and are "Keyboard-focusable", like these:

![CleanShot 2024-09-23 at 10 05 42@2x](https://github.com/user-attachments/assets/0d5c8612-6080-4afe-98c6-8ff4fc12a022)

![CleanShot 2024-09-23 at 10 05 16@2x](https://github.com/user-attachments/assets/1a1cde80-2c74-4dd4-b7f2-bc990c973d52)

These two images are also Image Blocks added to pages in the Twenty Twenty-Four theme.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

Report: 8762113-zen (this report contains links to a video and more details in a forum post)

I reviewed the site using Microsoft Edge Dev v130.0.2835.2 on macOS 14.7

The customer who reported the issue was using Chrome on Windows 11. I have requested details of the screen reader they are using.

**Comment by cpapazoglou** [comment]
@pauljacobson is there a chance that those [images are links](https://www.w3.org/WAI/tutorials/images/functional/)? What happens if you click on them?

**Comment by pauljacobson** [comment]
> is there a chance that those [images are links](https://www.w3.org/WAI/tutorials/images/functional/)? What happens if you click on them?

Hi. No, the images don't have links applied to them. Take a look at this one for example:

![CleanShot 2024-09-23 at 11 05 45@2x](https://github.com/user-attachments/assets/4757f773-56a5-42b6-8927-1e235a7bbc5c)

**Comment by lizswafford** [comment]
Also reported here: 8755557-zd (same site). 

The site logo (site icon) is set to not link to the home page but is still assigned the attribute role="button", which is picked up by the screen reader. 

On the services page, the images in the Media and Text Block also show as clickable even though no link has been assigned.
I used this CSS to test which images on a page are affected:
 [role="button"] {
    display: none !important;
}

**Comment by monsieur-z** [comment]
> On the services page, the images in the Media and Text Block also show as clickable even though no link has been assigned.

<img width="1431" alt="Screenshot 2024-09-25 at 3 05 46 PM" src="https://github.com/user-attachments/assets/abaa901b-0cae-4b2e-b14b-bdfefebe2996">

These images also have the attribute `role="button"`. Additionally, `tabindex=0` makes them navigable with a keyboard, though they're not interactive elements. Those two attributes should be likely be removed.

If the user wants these images to be strictly decorative, the `alt` attribute should be empty (not removed).

**Comment by mxhassani** [comment]
**For a workaround we can use:**

1. Upgrade them to the Business plan.
2. Add this plugin: https://wordpress.com/plugins/custom-css-js/
3. Then add this code to a JS snippet with that plugin:

```
document.querySelectorAll("img").forEach(function(img) {
  if (!img.closest('a')) {
    img.setAttribute('role', 'img');
  }
});
```
 4. Make sure to select "Where on page > In the <footer> element"

**Comment by Robertght** [comment]
I updated the labels here too.

@cpapazoglou do you need PA's help here in any way moving forward? 

I can see @mxhassani seems like a quick fix, so I wonder if we can use that.

**Comment by cpapazoglou** [comment]
@Robertght the behaviour is related to Jetpack Image Gallery Carousel feature ( [code](https://github.com/Automattic/jetpack/blob/f3abe0a43de003d0d3ab497d6fcc9192f4b28b3e/projects/plugins/jetpack/modules/carousel/jetpack-carousel.php#L997) ). 

Disabling the feature from `/wp-admin/options-media.php` doesn't add the `role` attribute at all ( same as core ).

![Image](https://github.com/user-attachments/assets/0afbacf3-2f73-4a9c-9504-7b1256a78b03)

The change has been introduced in https://github.com/Automattic/jetpack/pull/37792/files#diff-20a0b22f28fd1f2b380b5df535f70d1bd90cca42606e3ff5c7d57fba2ffcbfc2R997

I don't think that the `role = button` is the problem in this case, since if it worked as expected the image would be clickable and would open in fullscreen carousel. The problem is that in my tests doesn't work as expected ( [example post](https://cpappremium.wordpress.com/2024/10/11/test-2/) ). Which means, the carousel code should either not run at all OR it should behave as expected ( opening the image in a carousel ). 

@coder-karen is this something that your team can wrangle?

**Comment by zaguiini** [comment]
It seems that the reason why we're seeing the image attribute changes despite no carousel being loaded is because we're adding [this filter](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/carousel/jetpack-carousel.php#L123) regardless if there's a carousel on the page.

We need to guard it, or remove it entirely, only adding the filter once we're sure a carousel's been introduced.
